### PR TITLE
BUG: Execute window functions on joins

### LINKED ISSUE #1207: BUG: mutate with a joined scalar
# working
t1.mutate(foo=t1.ints.sum()).execute()

# failing
joined.mutate(sum=joined.value.sum()).execute()

### PR DESCRIPTION
**Pull request by cpcloud**
@jreback This fixes the bug we talked about earlier today.

Closes #1207

### CODE REVIEW

File: ibis/expr/types.py [hunk 1]
```diff
@@ -648,11 +648,12 @@ def param(type, name=None):
     return expr.name(name)
 
 
-def distinct_roots(*args):
-    all_roots = []
-    for arg in args:
-        all_roots.extend(arg._root_tables())
-    return list(toolz.unique(all_roots, key=id))
+def distinct_roots(*expressions):
```
Comment (jreback) [inline ibis/expr/types.py]: isn't this very similar to ``physical_tables``?

Comment (cpcloud) [reply ibis/expr/types.py]: Yes except that `physical_tables` will flatten some selection expressions, this doesn't

Comment (jreback) [reply ibis/expr/types.py]: ok, maybe add a doc-string to that effect? also should these be next to each other?

Comment (cpcloud) [reply ibis/expr/types.py]: I'll merge this PR, then push a commit to master with this docstring change.

**Pull request closed without merging**